src/Avalonia.Visuals/Media/Rotate3DTransform.cs

```diff
@@ -0,0 +1,201 @@
+    /// </summary>
+    /// <param name="angleX">The rotation around the X-Axis</param>
+    /// <param name="angleY">The rotation around the Y-Axis</param>
+    /// <param name="angleZ">The rotation around the Z-Axis</param>
+    /// <param name="centerX">The origin of the X-Axis</param>
+    /// <param name="centerY">The origin of the Y-Axis</param>
+    /// <param name="centerZ">The origin of the Z-Axis</param>
+    /// <param name="depth">The depth of the 3D effect</param>
+    public Rotate3DTransform(
+        double angleX,
+        double angleY,
+        double angleZ,
+        double centerX,
+        double centerY,
+        double centerZ,
+        double depth) : this()
+    {
+        _isInitializing = true;
+        AngleX = angleX;
+        AngleY = angleY;
+        AngleZ = angleZ;
+        CenterX = centerX;
+        CenterY = centerY;
+        CenterZ = centerZ;
+        Depth = depth;
+        _isInitializing = false;
+    }
+
+    /// <summary>
+    /// Sets the rotation around the X-Axis
+    /// </summary>
+    public double AngleX
+    {
+        get => GetValue(AngleXProperty);
+        set => SetValue(AngleXProperty, value);
+    }
+
+    /// <summary>
+    /// Sets the rotation around the Y-Axis
+    /// </summary>
+    public double AngleY
+    {
+        get => GetValue(AngleYProperty);
+        set => SetValue(AngleYProperty, value);
+    }
+
+    /// <summary>
+    ///  Sets the rotation around the Z-Axis
+    /// </summary>
+    public double AngleZ
+    {
+        get => GetValue(AngleZProperty);
+        set => SetValue(AngleZProperty, value);
+    }
+
+    /// <summary>
+    ///  Moves the origin the X-Axis rotates around
+    /// </summary>
+    public double CenterX
+    {
+        get => GetValue(CenterXProperty);
+        set => SetValue(CenterXProperty, value);
+    }
+
+    /// <summary>
+    ///  Moves the origin the Y-Axis rotates around
+    /// </summary>
+    public double CenterY
+    {
+        get => GetValue(CenterYProperty);
+        set => SetValue(CenterYProperty, value);
+    }
+
+    /// <summary>
+    ///  Moves the origin the Z-Axis rotates around
+    /// </summary>
+    public double CenterZ
+    {
+        get => GetValue(CenterZProperty);
+        set => SetValue(CenterZProperty, value);
+    }
+
+    /// <summary>
+    ///  Affects the depth of the rotation effect
+    /// </summary>
+    public double Depth
+    {
+        get => GetValue(DepthProperty);
+        set => SetValue(DepthProperty, value);
+    }
+
+    /// <summary>
+    /// Gets the transform's <see cref="Matrix"/>. 
+    /// </summary>
+    public override Matrix Value
+    {
+        get
+        {
+            var matrix44 = Matrix4x4.Identity;
+            var centerSum = CenterX + CenterY + CenterZ;
+            
+            if (centerSum != 0) matrix44 *= Matrix4x4.CreateTranslation(-(float)CenterX, -(float)CenterY, -(float)CenterZ);
```

Comment:
Comparing floats to 0 is especially not a good idea if make a sum before, so it may be better to take the typical compare approach: `Math.Abs(CenterSum) > double.Epsilon`. I am not sure if this is really here a problem as this would just solve rare performance edge cases which many transformations.

Reply:
Thanks, totally makes sense. Updated the code!